dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -292,11 +292,24 @@ bool IsLifetimeManagerViaEnumeration()
         return true;
     }
 
-    // Enable the Enumeration-style LifetimeManager if the environment variable
-    // MICROSOFT_WINDOWSAPPRUNTIME_DDLM_ENUMERATION is defined (for testing scenarios)
-    if (GetEnvironmentVariableW(L"MICROSOFT_WINDOWSAPPRUNTIME_DDLM_ENUMERATION", nullptr, 0) > 0)
+    // Select the LifetimeManager implementation if the environment variable
+    // MICROSOFT_WINDOWSAPPRUNTIME_DDLM_ALGORITHM is defined (for testing scenarios)
+    // where:
+    //     envvar=0 => Enumeration
+    //     envvar=1 => AppExtension
     {
-        return true;
+        WCHAR value[1 + 1]{};
+        if (GetEnvironmentVariableW(L"MICROSOFT_WINDOWSAPPRUNTIME_DDLM_ALGORITHM", value, ARRAYSIZE(value)) > 0)
```

Comment:
>GetEnvironmentVariableW [](http://example.com/codeflow?start=12&length=23)

If lpBuffer is not large enough to hold the data, the return value is the buffer size, in characters, required to hold the string and its terminating null character and the contents of lpBuffer are undefined.